### DISCUSSION TITLE
Keysight vna P5004B via thunderbolt

### DISCUSSION THREAD

**Discussion by kaspergr** [microsoft/Qcodes#5992] [Q&A]
Hi Community,

Any hints (or does a driver exist) on communicating with a Keysight Vector Network Analyzer P5004B. This model only communicates via a thunderbolt connection. We tried a fast attempt using the generic Keysight driver N52xx (KeysightP5004B calls this driver). 

The PXI address is obtained from the Keysight connection expert and the instrument works using Keysight software.

vna = KeysightP5004B('VNA', address = 'PXI0::CHASSIS1::SLOT1::FUNC0::INSTR')

This gives rise to a communication error
Could not connect at PXI0::CHASSIS1::SLOT1::FUNC0::INSTR
...
raise TypeError("QCoDeS only support MessageBasedResource Visa resources")
TypeError: QCoDeS only support MessageBasedResource Visa resources

Best regards, 
Kasper

**Comment by jenshnielsen** [comment]
Hi Kasper,

The logic that you stupled over is here https://github.com/microsoft/Qcodes/blob/4db9de3eafbcb9652751c3a9948e55db22f3b812/src/qcodes/instrument/visa.py#L189

I do not remember exactly why the check is there but it may simply be that we did not have access to any None Message based resources so I would suggest that you try to simply remove it and see how far that will get you.

It may also be worthwhile to experiment with pyvisa to see if you can connect with that directly https://pyvisa.readthedocs.io/en/latest/

**Comment by kaspergr** [comment]
Hi Henrik,

Thanks a lot for getting back to us so fast. We will try these two things.
We are also in contact with Keysight on the communication protocol they use.

Best,
Kasper

Fra: Jens Hedegaard Nielsen ***@***.***>
Sendt: 23. april 2024 08:30
Til: microsoft/Qcodes ***@***.***>
Cc: Kasper Grove-Rasmussen ***@***.***>; Author ***@***.***>
Emne: Re: [microsoft/Qcodes] Keysight vna P5004B via thunderbolt (Discussion #5992)


Hi Kasper,

The logic that you stupled over is here https://github.com/microsoft/Qcodes/blob/4db9de3eafbcb9652751c3a9948e55db22f3b812/src/qcodes/instrument/visa.py#L189

I do not remember exactly why the check is there but it may simply be that we did not have access to any None Message based resources so I would suggest that you try to simply remove it and see how far that will get you.

It may also be worthwhile to experiment with pyvisa to see if you can connect with that directly https://pyvisa.readthedocs.io/en/latest/
